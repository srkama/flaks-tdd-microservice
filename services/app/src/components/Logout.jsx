import React, { Component } from 'react'

export default class Logout extends Component {
    componentDidMount() {
        this.props.logout()
    }

    render() {
        return (
            <p>You are logged out!</p>
        )
    }
}
