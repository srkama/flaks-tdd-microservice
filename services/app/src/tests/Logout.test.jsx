import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Logout from '../components/Logout';

const logoutUser = jest.fn();

test('Logout renders properly', () => {
    const wrapper = shallow(<Logout logout={logoutUser} />);
    const element = wrapper.find('p')
    expect(element.length).toBe(1);
    expect(element.get(0).props.children).toBe('You are logged out!');
})

test('Logout renders snapshot properly', () => {
    const tree = renderer.create(<Logout logout={logoutUser} />).toJSON();
    expect(tree).toMatchSnapshot();
})
