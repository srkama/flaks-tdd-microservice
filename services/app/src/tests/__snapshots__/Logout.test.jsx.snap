// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders snapshot properly 1`] = `
<p>
  You are logged out!
</p>
`;
